packit/local_project.py

```diff
@@ -123,6 +123,25 @@ def __init__(
                 self.working_dir_temporary = True
                 change = True
 
+            if self.git_repo and not self.git_url:
+                # this is prone to errors
+                # also if we want url to upstream, we may want to ask for it explicitly
+                # since this can point to a fork
+                # .urls returns generator
+                self.git_url = list(self.git_repo.remote().urls)[0]
+                change = True
+
+            if self.git_url and not self.namespace:
+                if self.git_url.startswith("http"):
+                    # if git_url is in format http{s}://github.com/org/repo_name
+                    self.namespace = self.git_url.split("/")[3]
+                    change = True
+                else:
+                    # If git_url is in format git@github.com:org/repo_name
+                    org_repo = self.git_url.split(":")[1]
+                    self.namespace = org_repo.split("/")[0]
```

Comment:
I would move the parsing logic to the dedicated function. (Here or in the `packit/utils.py`.) 

- You can return both `namespace` and `repo` as a tuple.
- You can create unit tests for the parsing and with [parametrize](https://github.com/packit-service/packit/blob/master/tests/unit/test_config.py#L19) we can test multiple URL examples.